diff --git a/esm/index.js b/esm/index.js
index 7e6b46912faec6e577311c626683c97211ed34fb..e8c4c466dd09281ab33b0b301cbde0e209ffe89a 100644
--- a/esm/index.js
+++ b/esm/index.js
@@ -4,8 +4,14 @@ export { Ono } from "./constructor";
 export * from "./types";
 export { ono };
 export default ono;
+/*
+ * Commenting the rest of the file out because it causes problems with Vite.
+ * Since this is an ESM file, module should be undefined. Somehow, it ends up being defined with `exports` as an empty object.
+ * This causes the `Object.assign` call to fail, because `module.exports.default` is undefined.
+ * Ultimately, this should not affect a consumer of this module because there is a default export above.
+ */
 // CommonJS default export hack
-if (typeof module === "object" && typeof module.exports === "object") {
-    module.exports = Object.assign(module.exports.default, module.exports);
-}
+// if (typeof module === "object" && typeof module.exports === "object") {
+//     module.exports = Object.assign(module.exports.default, module.exports);
+// }
 //# sourceMappingURL=index.js.map